### DISCUSSION TITLE
Add support for Read Indicators in MessageListView

### DISCUSSION THREAD

**Discussion by mruth1022** [GetStream/stream-chat-android#4656] [Ideas]
Similar to the functionality provided by the iOS SDK ([docs here](https://getstream.io/chat/docs/sdk/ios/swiftui/message-components/read-indicators/)), we have a need to show read indicators (i.e. read receipts) within the list of messages in a channel (provided by the MessageListView UI component).

Ideally this would provide information on:
- whether or not the message was read by another user (other than the current user)
- the names of the users who have read the message

This information should be updated live through the web socket connection, such that it can provide real-time updates on when a message has been read and by whom.

Looking through the documentation for the Android SDK, I've seen that there is limited support for the read status of the last message of the channel, provided by ChannelListView ([docs here](https://getstream.io/chat/docs/sdk/android/ui/channel-components/channel-list/)). However, this is not sufficient as it doesn't indicate who has read the message, nor provide the read status for other messages in the channel besides the last message.

**Comment by filbabic** [comment]
Hey @mruth1022 !

As far as the state and UI goes, we already support:
* Channel.read state that provides the info like iOS (I believe it's the same level of detail)
* Read receipts for channels (if it's been sent but not read or if it's been read)
* Live updates through WS

I'm not sure what's missing here - you can use the state for the read status and utilize that to see which messages have been read by whom.

Could you explain a bit more what specifically is missing?

**Reply by filbabic** [reply]
![image](https://user-images.githubusercontent.com/17215808/216576626-ea773aea-984a-4008-b82c-acd2766d9f02.png)

As you can see, our channel supports the read state.

If you look into what that means, we have the following:

```kotlin
public data class ChannelUserRead(
    override var user: User,
    var lastRead: Date? = null,
    var unreadMessages: Int = 0,
    var lastMessageSeenDate: Date? = null
) : UserEntity
```

So we expose for each channel, a list of all the people who have read a message (any), which user it is, when the last message they read happened and how many unread messages they have.

**Comment by andremion** [comment]
Read indicators are supported: https://github.com/GetStream/stream-chat-android/blob/main/stream-chat-android-ui-components/src/main/res/values/attrs_message_list_view.xml#L313
